analyzers/src/SonarAnalyzer.Common/Rules/SymbolicExecutionRunnerBase.cs

```diff
@@ -62,17 +73,21 @@ protected void Analyze(SonarAnalysisContext sonarContext, SyntaxNodeAnalysisCont
             {
                 var isTestProject = sonarContext.IsTestProject(nodeContext.Compilation, nodeContext.Options);
                 var isScannerRun = sonarContext.IsScannerRun(nodeContext.Options);
-                AnalyzeSonar(nodeContext, isTestProject, isScannerRun, body, symbol);
-                if (ControlFlowGraph.IsAvailable)   // ToDo: Make this configurable for UTs when migrating other rules, see DeadStores
+                if (useSonarCfg)
+                {
+                    AnalyzeSonar(nodeContext, SonarRules, isTestProject, isScannerRun, body, symbol);
+                }
+                else
```

Comment:
This isn't the correct granularity of the decision. In production, we still need to run most of the rules under "Sonar" and some under "Roslyn" at the same time.
You should revert all these changes around rule registering, to get `SonarRules` and `AllRules` back. Keep the constructors and `useSonarCfg` flag.

From that, we'll probably need some temporary intermediate state before we rework the way `RuleFactory` works. I'm not sure if we'll have to rework it in this PR or if we can do some hacky workaround here and proper implementation there.

In another intermediate step, we still need the rules using Sonar-only for Peach and ITs to use Sonar-only.

At the final state:
Inside `AnalyzeSonar`, there should be something like `SonarRules.Where(..) concat AllRules that have Sonar version IF we are under UseSonarCfg state`

Inside `AnalyzeRoslyn` will be similar counterpart.

Reply:
For the NullPointerDereferenceTests I need to instruct the runner to use either "Roslyn" or "Sonar:

https://github.com/SonarSource/sonar-dotnet/blob/e6ee4b9d53df85372950553d506d6ce076f944de/analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/NullPointerDereferenceTest.cs#L29-L37

Both need to be kept in the future, because "Soanar" is executed in production if "cfg" is not available.
My assumption here was:
* Roslyn is the default
* Any "Sonar" is either opt-in for tests or fallback if cfg is not available.
* In the Roslyn case, we need as an intermediate solution another fallback:
  * Any DiagnosticId not yet migrated from Sonar to Roslyn gets executed with the Sonar engine. (That is the "NonMigratedRules" in line 82).

Therfore, I made "sonar" opt-in for tests and changed all occurrences to make this explicit (this was some extra effort needed at one point, when the fallback Roslyn->Sonar was not yet implemented. It isn't needed at the current state, but it will be needed once we start migrating the single rules over and run them for both engines).

> In another intermediate step, we still need the rules using Sonar-only for Peach and ITs to use Sonar-only.

I don't get this part. Do we need a new flag for this or can we re-use the existing one. What flag is set to what value on peach at the moment?

The ITs are passing at the moment. I don't see what additional need we have here.